src/js/AppServicesNav.js

```diff
@@ -173,23 +194,26 @@ class AppServicesNav extends React.Component {
                         onClick={() => this.setActive("weather")}>
                         <WeatherButton temp={weatherData.currentTemperature} weatherIcon={weatherData.weatherIcon}/>
                         <div className={"t-small t-medium th-f-color tab-text even-spacing " + ((activeServiceTab == "weather") ? "" : "hide-tab")}>
-                            <div className="weather-item">
+                            <p className={weatherMissingData ? "waiting-message" : "hide-tab"}>
+                                {weatherMissingData ? waitingMessage : null}
+                            </p>
+                            <div className={weatherMissingData ? "hide-tab" : "weather-item"}>
```

Comment:
`weatherData.high && weatherData.low` should probably be part of this condition, otherwise there is just a blank space before the rest of the data.